cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1687,10 +1684,9 @@ elaborateInstallPlan verbosity platform compiler compilerprogdb pkgConfigDB
         elabRegisterPackageDBStack = buildAndRegisterDbs
 
         elabSetupScriptStyle       = packageSetupScriptStyle elabPkgDescription
-        -- Computing the deps here is a little awful
-        deps = fmap (concatMap (elaborateLibSolverId mapDep)) deps0
```

Comment:
I'm not completely sure I calculated the closure of the setup dependencies correctly, since I skipped calling `elaborateLibSolverId mapDep`.  I called `CD.setupDeps` on `deps0` directly, because all setup dependencies are library dependencies.